api/swagger-batch.yaml

```diff
@@ -0,0 +1,310 @@
+openapi: 3.0.3
+info:
+  title: Turing Batch Experiment API
+  version: 0.0.1
+servers:
+  - url: localhost:8080/v1
+paths:
+  "/projects/{project_id}/ensemblers":
+    get:
+      tags: [ "Ensembler" ]
+      summary: Returns a list of ensemblers that belong to the project
+      parameters:
+        - in: path
+          name: project_id
+          schema:
+            $ref: "#/components/schemas/Id"
+          required: true
+        - in: query
+          name: page
+          schema:
+            type: integer
+            default: 1
+        - in: query
+          name: page_size
+          schema:
+            type: integer
+            default: 10
+      responses:
+        "200":
+          description: A JSON object, that represents paginated results response
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: "#/components/schemas/PaginatedResults"
+                  - type: object
+                    properties:
+                      results:
+                        type: array
+                        items:
+                          $ref: "#/components/schemas/GenericEnsembler"
+    post:
+      tags: [ "Ensembler" ]
+      summary: Saves a user-defined ensembler into the project
+      parameters:
+        - in: path
+          name: project_id
+          schema:
+            $ref: "#/components/schemas/Id"
+          required: true
+      requestBody:
+        description: A JSON object containing information about the ensembler
+        required: true
+        content:
+          application/json:
+            schema:
+              oneOf:
+                - $ref: "#/components/schemas/PyFuncEnsembler"
+      responses:
+        "201":
+          description: A JSON representation of a successfully saved ensembler
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: "#/components/schemas/PyFuncEnsembler"
+
+  "/projects/{project_id}/ensemblers/{ensembler_id}":
```

Comment:
Yeah, it is intentional. I wasn't sure what subsequences will it have if a user removes the ensembler, that was used for submitting an ensembling job (the API for that is WIP by @ashwinath), so I skipped the deletion for now

Reply:
The complication would arise when the job has been submitted and running/queued up since it's running in the background. Extra check logic needs to be done to ensure that the job should run if the ensembler is missing. While this can be done, I would suppose that it is a matter of whether it is worth the time implementing such a feature.

Reply:
For the routers / versions, we have additional checks to ensure there are no other references to an entity we are trying to delete (or, in some cases, all associated entities are deleted together):
* Can't delete currently deployed router version
*  Can't delete a router version if that's the only version under a router (the router must be deleted directly)
* ...
So, to @ashwinath's point on "Extra check logic needs to be done to ensure that the job should run if the ensembler is missing." - perhaps the delete can be disalowed if there is a job in progress referencing it.

Anyway, sounds like Delete has not been added yet for these considerations. Thanks for the clarification.

Reply:
The other way of dealing with it would be to treat records from `ensemblers` table some sort of templates, and create a snapshot config from these templates at the deployment time (either router deployment / job submission). So in this case, the actual record from the `ensemblers` table can be safely deleted at any time, because the job will use a copy of it's data.
But I'll keep it as is for now and get back to this question once the jobs API is ready